### DISCUSSION TITLE
schemaValidator vs onValidate

### DISCUSSION THREAD

**Discussion by simplecommerce** [vazco/uniforms#1136] [Q&A] [answered]
Hi, just a quick question.

Since `onValidate` can be used for both sync and async validation, what would be the point to use `schemaValidator` for sync validations instead of just adding them all in `onValidate` ?

I am just wondering, since in my project I am trying to simplify things for the devs, I was asking myself this.

Thanks!

**Comment by radekmie** [comment, marked as answer]
Hi @simplecommerce. What kind of `schemaValidator` you have in mind? The one in GraphQL or JSON Schema bridge? If so, the only reason why I'd still leave them separated is convenience. The bridge always defines some validator, and it can be further extended in a form.

We _may_ want to sail off this idea in the next versions, but for now, I don't see it as a problem. In the end, most users never use `onValidate` nor define the validator themselves (i.e., their bridge does it automatically, or they use `ajv` or other library).

**Reply by simplecommerce** [reply]
@radekmie Sorry about that, I am using GraphQL.

In my case I implemented a modified version of the bridge to auto validate required fields, so I rarely use the `schemaValidator`, but do use `onValidate` for async validation.

So I assume since its only for convenience, there shouldn't be any issues if I only use `onValidate` for all my validation sync and async.

**Reply by radekmie** [reply]
No, there's no problem with that. I do, however, encourage everyone implementing their own bridges to keep most of the schema-dependent logic in the bridge, and use form props (like `onValidate`) only for form-specific code. This way a potential change in the library will require fewer changes in your project.
